ykcompile/src/lib.rs

```diff
@@ -282,15 +276,26 @@ impl<TT> TraceCompiler<TT> {
         }
     }
 
+    /// Returns a free register or `None` if all registers are occupied.
+    fn get_free_register(&self) -> Option<u8> {
+        self.register_content_map.iter().find_map(|(k, v)| match v {
+            RegisterVal::Free => Some(*k),
+            _ => None,
+        })
+    }
+
+    /// Spill a local to the stack and return its location.
```

Comment:
Should this also update the mapping from locals to locations?

Reply:
I guess it could. I left it this way due to the way it's used in https://github.com/softdevteam/yk/pull/104/files/6b48c0933870c5148cfd3a8d3c2416687cb50ef6..43ad10f19a2ac4eeb8419734c38affc40af7d86e#diff-162e1dfdca14767450c768ad342d54f2R265. So this would need some more refactoring.

Reply:
For now let's add to the comment saying that it doesn't actually update the mapping?